scanpy/plotting/_tools/scatterplots.py

```diff
@@ -369,7 +370,7 @@ def embedding(
                     in_groups_size = not_in_groups_size = size
 
                 # only show grey points if no image is below
-                if img_key is None:
+                if library_id is None:
```

Comment:
What does this branch do? It seems like the next statement does the same thing?

Reply:
when groups are defined, then only cells in that group are colored, the others are light gray. When `sc.pl.spatial` is plotted, we only want spots in the specific groups, and the others should simply not be plotted. 
You can have a look at cell 12 in the [tutorial](https://scanpy-tutorials.readthedocs.io/en/latest/spatial/basic-analysis.html). If this if statement is not passed, gray spots would be plotted instead of leaving it blank.

Reply:
Ah, I see. I was missing the `~`